src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java

```diff
@@ -77,10 +92,10 @@ public void write(RevObject o, OutputStream out) throws IOException {
         final int blockSize = 1 << 16;
         LZ4Compressor compressor = lz4factory.fastCompressor();
         Checksum checksum = newChecksum();
-        try (LZ4BlockOutputStream cout = new LZ4BlockOutputStream(out, blockSize, compressor,
-                checksum, false)) {
-            factory.write(o, cout);
-        }
+        LZ4BlockOutputStream cout = new LZ4BlockOutputStream(out, blockSize, compressor, checksum,
+                true);
+        factory.write(o, cout);
+        cout.finish();// same as close but not closing the wrapped stream
```

Comment:
should this be in a finalize?

Reply:
I don't think so, if factory.write(o, cout) throws an exception then it's propagated and nothing can really be done so there's no point in forcing to flush the compression buffer